pandas/tests/io/test_sql.py

```diff
@@ -4358,3 +4358,21 @@ def test_xsqlite_if_exists(sqlite_buildin):
         (5, "E"),
     ]
     drop_table(table_name, sqlite_buildin)
+
+
+@pytest.mark.parametrize("dtype_backend", ["pyarrow", "numpy_nullable", lib.no_default])
+def test_bytes_column(sqlite_buildin, dtype_backend):
+    pa = pytest.importorskip("pyarrow")
+    """
+    Regression test for (#59242)
+    Bytes being returned in a column that could not be converted
+    to a string would raise a UnicodeDecodeError
+    when using dtype_backend='pyarrow' or dtype_backend='numpy_nullable'
+    """
+    query = """
+    select cast(x'0123456789abcdef0123456789abcdef' as blob) a
+    """
+    df = pd.read_sql(query, sqlite_buildin, dtype_backend=dtype_backend)
+    assert df.a.values[0] == b"\x01#Eg\x89\xab\xcd\xef\x01#Eg\x89\xab\xcd\xef"
```

Comment:
Can you use our built-in test helpers instead? I think you can just do:

```python
result = pd.read_sql(...)
expected = pd.DataFrame({"a": ...}, dtype=pd.ArrowDtype(pa.binary()))
tm.assert_frame_equal(result, expected)
```

What data type does this produce currently with the `numpy_nullable` backend - object?

Reply:
for sure, changed the testing logic over to using this!  for `numpy_nullable` and `lib.no_default` the dtype returned is an object